src/components/body/BodyMainPanel.tsx

```diff
@@ -45,20 +57,76 @@ const BodyMainPanel: React.FC<IProps> = ({
 
   const contentContainerStyle = React.useMemo(
     () => ({
-      paddingLeft: styleLeft,
-      paddingTop: styleTop,
+      top: styleTop,
+      left: styleLeft,
+      paddingLeft: _scrollLeft,
+      paddingTop: _scrollTop,
       height: dataLength * bodyRowHeight,
       width: bodyContentWidth,
     }),
-    [styleTop, styleLeft, dataLength, bodyRowHeight, bodyContentWidth]
+    [
+      styleTop,
+      styleLeft,
+      _scrollTop,
+      _scrollLeft,
+      dataLength,
+      bodyRowHeight,
+      bodyContentWidth,
+    ]
   );
 
+  const throttledScroll = React.useMemo(
+    () =>
+      throttle((scrollTop: number, scrollLeft: number) => {
+        console.log("throttled");
+        if (scrollTop < 0) scrollTop = 0;
+        else if (scrollLeft < 0) scrollLeft = 0;
+        else if (scrollTop > _bodyHeight)
+          scrollTop = dataLength * bodyRowHeight;
+        else if (scrollLeft > _bodyWidth - lineNumberColumnWidth)
+          scrollLeft = _bodyWidth - lineNumberColumnWidth;
```

Comment:
Divide calculation for scrollTop and scrollLeft to each part.

if ~ else if  <- For Top

if ~ else if  <- For Left